src/ripple/app/misc/impl/Transaction.cpp

```diff
@@ -99,28 +100,29 @@ Transaction::pointer Transaction::transactionFromSQL (
     return tr;
 }
 
-Transaction::pointer Transaction::transactionFromSQLValidated(
-    boost::optional<std::uint64_t> const& ledgerSeq,
-    boost::optional<std::string> const& status,
-    Blob const& rawTxn,
-    Application& app)
+Transaction::pointer Transaction::load (uint256 const& id, Application& app, error_code_i& ec)
 {
-    auto ret = transactionFromSQL(ledgerSeq, status, rawTxn, app);
+    using op = boost::optional<ClosedInterval<uint32_t>>;
 
-    if (checkValidity(app.getHashRouter(),
-            *ret->getSTransaction(), app.
-                getLedgerMaster().getValidatedRules(),
-                    app.config()).first !=
-                        Validity::Valid)
-        return {};
+    return boost::get<pointer> (load (id, app, op {}, ec));
+}
 
-    return ret;
+boost::variant<Transaction::pointer, bool>
+Transaction::load (uint256 const& id, Application& app, ClosedInterval<uint32_t> const& range,
+    error_code_i& ec)
+{
+    using op = boost::optional<ClosedInterval<uint32_t>>;
+
+    return load (id, app, op {range}, ec);
```

Comment:
No need the type alias and `op{range}`. Just `range` will work (a temporary `boost::optional` will be created for the parameter).

Reply:
It turns out that just using `range` doesn't work because the compiler will end up calling the current method which will recurse infinitely.

Reply:
Oh, I missed that! Good catch. Yup, you need the wrapper.